lhotse/workflows/faster_whisper.py

```diff
@@ -0,0 +1,244 @@
+        cost of increased memory usage.
+    :param vad_filter: If True, use faster-whisper's built-in voice activity detection (SileroVAD).
+    :param add_alignments: if True, add word alignments using timestamps obtained using the cross-
+        attention pattern and dynamic time warping (Note: Less accurate than forced alignment).
+    :param decode_options: additional options to pass to the ``whisper.transcribe`` function.
+    :return: a generator of cuts (use ``CutSet.open_writer()`` to write them).
+    """
+    assert is_module_available("faster_whisper"), (
+        "This function expects faster-whisper to be installed. "
+        "You can install it via 'pip install faster-whisper' "
+        "(see https://github.com/guillaumekln/faster-whisper/ for details)."
+    )
+    if not isinstance(manifest, RecordingSet) and not isinstance(manifest, CutSet):
+        raise ValueError("The ``manifest`` must be either a RecordingSet or a CutSet.")
+    assert not vad_filter or is_module_available("onnxruntime"), (
+        "Use of VAD requires onnxruntime to be installed. "
+        "You can install it via 'pip install onnxruntime' "
+        "(see https://github.com/guillaumekln/faster-whisper/ for details)."
+    )
+    if vad_filter and add_alignments:
+        warnings.warn(
+            "Word timestamps can be very inaccurate when using VAD. We don't recommend using both "
+            f"options together. See https://github.com/guillaumekln/faster-whisper/issues/125."
+        )
+
+    model = _initialize_model(
+        model_name, device, compute_type, num_workers, download_root
+    )
+    with ThreadPoolExecutor(num_workers) as ex:
+        futures = []
+        for item in manifest:
+            futures.append(
+                ex.submit(
+                    _process_single_manifest,
+                    item,
+                    model,
+                    force_nonoverlapping,
+                    vad_filter,
+                    add_alignments,
+                    **decode_options,
+                )
+            )
+        for item in as_completed(futures):
+            yield item.result()
+
+
+def _initialize_model(
+    model_name: str,
+    device: str,
+    compute_type: str = "default",
+    num_workers: int = 1,
+    download_root: Optional[str] = None,
+):
+    import torch
+    from faster_whisper import WhisperModel
+
+    # Parse device index
+    device, _, idx = device.partition(":")
+    if len(idx) > 0:
+        device_index = int(idx)
+    elif num_workers > 1 and device == "cuda":
+        # Limit num_workers to available GPUs
+        num_workers = min(num_workers, torch.cuda.device_count())
+        device_index = list(range(num_workers))
+    else:
+        device_index = 0
+    model = WhisperModel(
+        model_name,
+        device=device,
+        device_index=device_index,
+        compute_type=compute_type,
+        num_workers=num_workers,
+        download_root=download_root,
```

Comment:
Since the change that enables this option is still not released in pip, I suggest a bit of workaround here, otherwise it cannot be ran:

```
+    opt_kwargs = {}
+    if download_root is not None:
+        opt_kwargs["download_root"] = download_root
     model = WhisperModel(
         model_name,
         device=device,
         device_index=device_index,
         compute_type=compute_type,
         num_workers=num_workers,
-        download_root=download_root,
+        **opt_kwargs,
     )
-    model.logger.setLevel(logging.WARNING)
+    if hasattr(model, "logger"):
+        model.logger.setLevel(logging.WARNING)
```

Note I also suggested a check for logger, on my installation `model` did not have the attribute `logger` defined.